webui/react/src/components/SlotAllocationBar.tsx

```diff
@@ -0,0 +1,159 @@
+import { Popover } from 'antd';
+import React, { useMemo } from 'react';
+
+import Badge from 'components/Badge';
+import Bar from 'components/Bar';
+import { getStateColorCssVar, ShirtSize } from 'themes';
+import { ResourceState } from 'types';
+import { ConditionalWrapper } from 'utils/react';
+import { floatToPercent } from 'utils/string';
+
+import { BadgeType } from './Badge';
+import css from './SlotAllocation.module.scss';
+
+export interface Props {
+  barOnly?: boolean;
+  className?: string;
+  resourceStates: ResourceState[];
+  showLegends?: boolean;
+  size?: ShirtSize;
+  totalSlots: number;
+}
+
+interface LegendProps {
+  children: React.ReactNode;
+  count: number;
+  totalSlots: number;
+  showPercentage?: boolean;
+}
+
+const Legend: React.FC<LegendProps> = ({
+  count, totalSlots,
+  showPercentage, children,
+}: LegendProps) => {
+
+  return (
+    <li className={css.legend}>
+      <span className={css.count}>
+        {count} {showPercentage && `(${floatToPercent(count/totalSlots, 0)})`}
+      </span>
+      <span>
+        {children}
+      </span>
+    </li>
+  );
+};
+
+const SlotAllocationBar: React.FC<Props> = ({
+  resourceStates,
+  totalSlots,
+  showLegends,
+  className,
+  ...barProps
+}: Props) => {
+
+  const stateTallies = useMemo(() => {
+    const tally: Record<ResourceState, number> = {
+      [ResourceState.Assigned]: 0,
+      [ResourceState.Pulling]: 0,
+      [ResourceState.Running]: 0,
+      [ResourceState.Starting]: 0,
+      [ResourceState.Terminated]: 0,
+      [ResourceState.Unspecified]: 0,
+    };
+    resourceStates.forEach(state => {
+      tally[state] += 1;
+    });
+    return tally;
+  }, [ resourceStates ]);
+
+  const freeSlots = (totalSlots - resourceStates.length);
+  const pendingSlots = (resourceStates.length - stateTallies.RUNNING);
+
+  const { barParts, legendParts } = useMemo(() => {
+
+    const parts = {
+      free: {
+        color: 'var(--theme-colors-monochrome-15)',
+        percent: freeSlots / totalSlots,
+      },
+      pending: {
+        color: '#6666CC',
```

Comment:
**chore:** let's add this to `StateColors` in `themes.ts` as `pending = 'pending',` and add `#6666cc` to `theme-colors-states-pending`.